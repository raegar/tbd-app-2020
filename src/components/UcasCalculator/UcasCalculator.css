.gradeTypeHeader {
    font-weight: bold
}

.qualInputBox {
    width: 100%;
}

.subjectInputBox {

}

.gradeSelect {

}

.addQualButton {

}

.qualPoints {

}

.totalQualPoints {

}