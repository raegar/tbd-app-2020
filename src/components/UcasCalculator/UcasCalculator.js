import React, {useState} from 'react';
import './UcasCalculator.css';
import {UCASGrades} from '../../global/Constants';
import TextBox from '../TextBox/TextBox';
import DropdownMenu from '../DropdownMenu/DropdownMenu';
import {Container, Row, Col} from "react-bootstrap";

function UcasCalculator() {
	//const [selectedGrades, setSelectedGrades] = useState([{}]);
	const [gradeOptions, setGradeOptions] = useState([]);

	function gradesOptions() {
		const options = [];

		for (const name in UCASGrades) {
			//first add the disabled bold section option
			options.push(<option disabled className="gradeTypeHeader">{name}</option>);

			//now add each subgrade type
			for (const subgradeName in UCASGrades[name]) {
				options.push(<option value={UCASGrades[name][subgradeName]}>{"  " + subgradeName}</option>);
			}
		}

		return options;
	}

	function selectGradeType(e) {
		console.log(JSON.stringify(e.target.value));
		setGradeOptions(e.target.value);
	}

	return (
		<div>
			<Container>
				<Row>
					<Col>
						<select id={"grades"} onChange={selectGradeType}>
							<option disabled selected="selected">Please select a qualification...</option>
							{gradesOptions()}
						</select>
					</Col>
				</Row>
				<Row>
					<Col>
						<TextBox placeholder="Subject (optional)"/>
					</Col>
					<Col>
						<DropdownMenu options={
							gradeOptions.map((option, i) => {
								return {
									label: option.grade,
									value: option.points
								};
							})
						}/>
					</Col>
				</Row>
			</Container>
		</div>
	);
}

//<iframe src="https://www.ucas.com/ucas/tariff-calculator"></iframe>
export default UcasCalculator;