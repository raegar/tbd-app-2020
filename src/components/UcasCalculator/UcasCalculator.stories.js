import React from 'react';
import {UcasCalculator} from './UcasCalculator';

export default {
    title: "UcasCalculator",
    component: UcasCalculator
}

export const Default = () => <UcasCalculator/>;