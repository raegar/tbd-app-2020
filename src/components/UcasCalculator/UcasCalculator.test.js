import React from 'react';
import { render, screen, cleanup, wrapper } from '@testing-library/react';
import '@testing-library/jest-dom';
import UcasCalculator from './UcasCalculator';

afterEach(cleanup);

it("Renders the calculator", () => {
    const { asFragment } = render(<UcasCalculator/>);
    expect(asFragment()).toMatchSnapshot();
});

test("Points 0 is displayed by default", () => {
    render(
        <UcasCalculator/>
    );

    expect(screen.getByTestId("qualPoints")).not.toBeNull();
    expect(screen.getByTestId("qualPoints")).toHaveTextContent("Points: 0");
});

test("Total points 0 is displayed by default", () => {
    render(
        <UcasCalculator/>
    );

    expect(screen.getByTestId("totalPoints")).not.toBeNull();
    expect(screen.getByTestId("totalPoints")).toHaveTextContent("0");//.toBe(0);
})

//TODO: add test for selecting a course and grade.