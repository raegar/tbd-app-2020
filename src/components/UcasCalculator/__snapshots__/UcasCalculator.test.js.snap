// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the calculator 1`] = `
<DocumentFragment>
  <div>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <select
            class="qualInputBox"
            id="qualifications"
          >
            <option
              disabled=""
            >
              Please select a qualification...
            </option>
            <option
              class="gradeTypeHeader"
              disabled=""
            >
              A Level and AS
            </option>
            <option
              value="A Level and AS;A Levels"
            >
                A Levels
            </option>
            <option
              value="A Level and AS;AS Levels"
            >
                AS Levels
            </option>
            <option
              class="gradeTypeHeader"
              disabled=""
            >
              BTEC National (RQF) (From 2016)
            </option>
            <option
              value="BTEC National (RQF) (From 2016);Pearson BTEC Level 3 National Extended Certificate (first teaching from September 2016)"
            >
                Pearson BTEC Level 3 National Extended Certificate (first teaching from September 2016)
            </option>
          </select>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-2"
        >
          <div
            class="mobiletextbox"
          >
            <label />
            <input
              class="subjectInputBox"
              maxlength="30"
              minlength="0"
              placeholder="Subject (optional)"
              size="20"
              type="text"
            />
          </div>
        </div>
        <div
          class="col-md-4"
        >
          <br />
        </div>
        <div
          class="col-md-2"
        >
          <br />
          <select
            class="Dropdown"
            disabled=""
          >
            <option
              value=""
            />
            <option
              value="none"
            >
              ...
            </option>
          </select>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-6"
        />
        <div
          class="col"
        >
          <span
            class="qualPoints"
            data-testid="qualPoints"
          >
            Points: 
            <b>
              0
            </b>
          </span>
        </div>
      </div>
      <br />
      <div
        class="row"
      >
        <button
          class="mediumbutton addQualButton"
          data-testid="ucpButton"
          type="button"
        >
          Add another qualificationâ€‹
        </button>
      </div>
      <span
        class="totalQualPoints"
      >
        Total points: 
        <b
          data-testid="totalPoints"
        >
          0
        </b>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
